server/szurubooru/func/posts.py

```diff
@@ -515,18 +555,30 @@ def update_post_content(post: model.Post, content: Optional[bytes]) -> None:
             and other_post.post_id != post.post_id:
         raise PostAlreadyUploadedError(other_post)
 
+    if update_signature:
+        purge_post_signature(post)
+        post.signature = generate_post_signature(post, content)
```

Comment:
Maybe we could do it periodically in that background thread instead? Then again, it'd cause the db to stay inconsistent…

Reply:
One nice aspect about the image uploads is that when I bulk upload variants of one image, the reverse search will immediately find a hit to the images just uploaded, allowing me to define a relation for them.

Reply:
Yeah I'm totally fine with this approach, as long as it doesn't have unreasonably long run time.